packages/ui5-config/test/index.test.ts

```diff
@@ -117,4 +117,49 @@ describe('UI5Config', () => {
         ui5Config.addCustomMiddleware([config], comments);
         expect(ui5Config.toString()).toMatchSnapshot();
     });
+
+    test('addCustomTask', () => {
+        ui5Config.addCustomTasks([
+            {
+                name: 'ui5-task-zipper',
+                afterTask: 'generateCachebusterInfo',
+                configuration: {
+                    archiveName: 'my-archive'
+                }
+            }
+        ]);
+        expect(ui5Config.toString()).toMatchSnapshot();
+    });
+
+    describe('addAbapDeployTask', () => {
+        const app: AbapApp = {
+            name: '~name',
+            desription: '~description',
+            package: '~package',
+            transport: '~transport'
+        };
+
+        test('local settings', () => {
+            ui5Config.addAbapDeployTask({ url, client }, app);
+            expect(ui5Config.toString()).toMatchSnapshot();
+        });
+
+        test('AppStudio + Steampunk settings', () => {
+            ui5Config.addAbapDeployTask(
+                {
+                    destination,
+                    scp: true
+                },
+                app
+            );
+            expect(ui5Config.toString()).toMatchSnapshot();
+        });
+    });
+
+    test('removeMiddleware', () => {
+        const { config, comments } = getFioriToolsProxyMiddlewareConfig([], {});
+        ui5Config.addFioriToolsAppReloadMiddleware();
```

Comment:
Could we use an already existing middleware that we delete instead? This adds a dependency to `addFioriToolsAppReloadMiddleware`. The problem with that is that bugs in that method could make this testcase fail. Or worse, bugs there could make this testcase pass when there's a bug in `removeMiddleware`.

Reply:
4bf4bcd